tests/async.rs

```diff
@@ -93,24 +93,35 @@ where
 
     fn test<C, F>(self, c: C)
     where
-        F: IntoFuture<Item = (), Error = mysql_async::errors::Error>,
-        C: FnOnce(mysql_async::Conn) -> F,
+        F: IntoFuture<Item = (), Error = mysql_async::error::Error> + 'static,
+        <F as futures::IntoFuture>::Future: std::marker::Send,
+        C: FnOnce(mysql_async::Conn) -> F + Send + 'static,
     {
+        let mut runtime = Runtime::new().unwrap();
+
         let listener = net::TcpListener::bind("127.0.0.1:0").unwrap();
         let port = listener.local_addr().unwrap().port();
+
         let jh = thread::spawn(move || {
             let (s, _) = listener.accept().unwrap();
+
             MysqlIntermediary::run_on_tcp(self, s)
         });
 
-        let mut core = Core::new().unwrap();
-        let handle = core.handle();
-        let db =
-            core.run(mysql_async::Conn::new(
-                &format!("mysql://127.0.0.1:{}", port),
-                &handle,
-            )).unwrap();
-        core.run(c(db).into_future()).unwrap();
+        let db_uri = format!("mysql://127.0.0.1:{}", port);
+
+        let pool = mysql_async::Pool::new(db_uri);
+
+        let future = pool.get_conn().and_then(|conn| {
+            c(conn)
+        });
+
+        let _res = runtime.block_on(future).unwrap();
```

Comment:
Why is the `let _res` necessary here? If what comes back is a result, we probably want another `.unwrap()` to make sure there indeed were no errors.

Reply:
Nope, another `.unwrap()` is much better.